quicktests/overlaying/tests/realistic/phones.js

```diff
@@ -0,0 +1,150 @@
+    {phoneType: "feature",  demographic: "Ages 35-44",  value: 0.86},
+    {phoneType: "feature",  demographic: "Ages 45-64",  value: 0.87},
+    {phoneType: "smart",    demographic: "Male",        value: 0.13},
+    {phoneType: "smart",    demographic: "Female",      value: 0.07},
+    {phoneType: "smart",    demographic: "Ages 16-24",  value: 0.14},
+    {phoneType: "smart",    demographic: "Ages 25-34",  value: 0.12},
+    {phoneType: "smart",    demographic: "Ages 35-44",  value: 0.09},
+    {phoneType: "smart",    demographic: "Ages 45-64",  value: 0.06}
+  ];
+
+  var dataRoK = [
+    {phoneType: "feature",  demographic: "Male",        value: 0.20},
+    {phoneType: "feature",  demographic: "Female",      value: 0.25},
+    {phoneType: "feature",  demographic: "Ages 16-24",  value: 0.08},
+    {phoneType: "feature",  demographic: "Ages 25-34",  value: 0.07},
+    {phoneType: "feature",  demographic: "Ages 35-44",  value: 0.21},
+    {phoneType: "feature",  demographic: "Ages 45-64",  value: 0.38},
+    {phoneType: "smart",    demographic: "Male",        value: 0.68},
+    {phoneType: "smart",    demographic: "Female",      value: 0.65},
+    {phoneType: "smart",    demographic: "Ages 16-24",  value: 0.86},
+    {phoneType: "smart",    demographic: "Ages 25-34",  value: 0.86},
+    {phoneType: "smart",    demographic: "Ages 35-44",  value: 0.66},
+    {phoneType: "smart",    demographic: "Ages 45-64",  value: 0.49}
+  ];
+
+
+  return [dataAustralia, dataIndia, dataRoK];
+}
+
+function run(svg, data, Plottable) {
+  "use strict";
+
+  var AusColor = "#ff6969";
+  var IndColor = "#662a48";
+  var RoKColor = "#cc3300";
+
+  var colorScale = new Plottable.Scales.Color();
+  colorScale.domain(["Australia", "India", "South Korea"]);
+  colorScale.range([AusColor, IndColor, RoKColor]);
+
+  var legend = new Plottable.Components.Legend(colorScale)
+    .maxEntriesPerRow(3)
+    .xAlignment("left");
+
+  var AusDataset = new Plottable.Dataset(data[0]);
+  var IndDataset = new Plottable.Dataset(data[1]);
+  var RoKDataset = new Plottable.Dataset(data[2]);
+
+  var xScale = new Plottable.Scales.Linear().domain([0, 1]);
+  var xScale_reverse = new Plottable.Scales.Linear().domain([1, 0]);
+  var yScale = new Plottable.Scales.Category();
+
+  var AusLabels = new Plottable.Axes.Category(yScale, "left")
+    .tickLength(0)
+    .endTickLength(0);
+  var IndLabels = new Plottable.Axes.Category(yScale, "left")
+    .tickLength(0)
+    .endTickLength(0);
+  var RoKLabels = new Plottable.Axes.Category(yScale, "left")
+    .tickLength(0)
+    .endTickLength(0);    
+  var featureLabel = new Plottable.Components.Label("Feature Phones", 0);
+  var smartLabel = new Plottable.Components.Label("Smart Phones", 0);
+
+  var filterFeatureX = function(d){
+    if(d.phoneType === "feature"){
+      return d.value;
+    }
+  };
+  var filterFeatureY = function(d){
+    if(d.phoneType === "feature"){
+      return d.demographic;
+    }
```

Comment:
The problem is that the function doesn't return anything if `d.phoneType !== "feature"`, so it returns `undefined`, which then gets stringified and put on the plot. A similar thing happens here:
http://jsfiddle.net/5ny4cs1r/1/